File: EcosystemSimulation/src/shmp/simulation/space/resource/dependency/Temperature.java
package shmp.simulation.space.resource.dependency;


public abstract class Temperature extends CoefficientDependency {
    int threshold;

    public Temperature(int threshold, double deprivationCoefficient) {
        super(deprivationCoefficient);
        this.threshold = threshold;
    }

    @Override
    public boolean isNecessary() {
        return true;
    }

    @Override
    public boolean isPositive() {
        return true;
    }

    @Override
    public boolean isResourceNeeded() {
        return false;
    }
}

File: EcosystemSimulation/src/io/tashtabash/sim/space/resource/dependency/Temperature.kt
package io.tashtabash.sim.space.resource.dependency


abstract class Temperature(
        var threshold: Int,
        deprivationCoefficient: Double
) : CoefficientDependency(deprivationCoefficient) {
    override val isNecessary = true

    override val isPositive = true

    override val isResourceNeeded = false
}
